common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -136,10 +136,17 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
                     public void onPageLoaded() {
                         mProgressBar.setVisibility(View.INVISIBLE);
 
-                        // Inject string from test suites.
-                        // if result already sent don't load url again
-                        if (mAuthResultSent == false && !StringExtensions.isNullOrBlank(mPostPageLoadedUrl)) {
-                            mWebView.loadUrl(mPostPageLoadedUrl);
+                        // Inject the javascript string from testing. This should only be evaluated if we haven't sent
+                        // an auth result already.
+                        if (mAuthResultSent == false && !StringExtensions.isNullOrBlank(mPostPageLoadedJavascript)) {
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+                                mWebView.evaluateJavascript(mPostPageLoadedJavascript, null);
+                            } else {
+                                // On earlier versions of Android, javascript has to be loaded with a custom scheme.
```

Comment:
Does the javascript scheme no longer work?

Reply:
It does work, but it has some interesting properties, which are described in the comment. I prefer to not escape data if it's not necessary.

Reply:
It's correct from a URL point of view.  Whatever you put in a URL should be URL-safe, so the parser *should* decode it before handing it to the js engine.  I was just curious how this came up and why we didn't notice it until now.

Reply:
It came up because one of our redirect URIs had a percent-encoded octet in it, and it was being passed into the js (as part of automation).